### PR TITLE
Adding PPE to known discovery endpoints

### PR DESCRIPTION
**Pull request by trwalke**
Fixes # 
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3265

**Changes proposed in this request**
Adding PPE to known instance discovery metadata

**Testing**

**Performance impact**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Instance/Discovery/KnownMetadataProvider.cs [hunk 1]
```diff
@@ -72,11 +72,19 @@ void AddToPublicEnvironment(InstanceDiscoveryMetadataEntry entry)
                 PreferredCache = "login-us.microsoftonline.com"
             };
 
+            InstanceDiscoveryMetadataEntry ppeCloudEntry = new InstanceDiscoveryMetadataEntry()
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Instance/Discovery/KnownMetadataProvider.cs:75]: The solution seems to be to simply add this entry to the known list. Seems like something we should have as it is well known for our first party customers? Thoughts @bgavrilMS ?

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Instance/Discovery/KnownMetadataProvider.cs:75]: Yes, this is correct.

File: tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/InstanceProviderTests.cs [hunk 2]
```diff
@@ -66,6 +66,10 @@ public void KnownMetadataProvider_RespondsIfEnvironmentsAreKnown()
                 "login.partner.microsoftonline.cn", new[] { "login.windows.net", "login.microsoft.com", "login.partner.microsoftonline.cn" }, _logger);
             Assert.IsNotNull(result);
 
+            result = knownMetadataProvider.GetMetadata(
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/InstanceProviderTests.cs:69]: Can we have a test around regional that confirms that when using PPE, an instance discovery call is not made ?

Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/InstanceProviderTests.cs:69]: You'll also need to fix the test `OtherCloudWithAuthorityValidationAsync`, as we still want to test with a "different cloud".

Comment (trwalke) [reply tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/InstanceProviderTests.cs:69]: > Can we have a test around regional that confirms that when using PPE, an instance discovery call is not made ?

Does this test already validate this?

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/5e29d349dde99775bf60f23812e2a2437bb0cca9/tests/Microsoft.Identity.Test.Unit/PublicApiTests/ClientCredentialWithRegionTests.cs#L398

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/ClientCredentialWithRegionTests.cs [hunk 3]
```diff
@@ -340,7 +340,7 @@ public async Task OtherCloudWithAuthorityValidationAsync()
 
                 var tokenHttpCallHandler = new MockHttpMessageHandler()
                 {
-                    ExpectedUrl = "https://eastus.login.windows-ppe.net/17b189bc-2b81-4ec5-aa51-3e628cbc931b/oauth2/v2.0/token",
+                    ExpectedUrl = "https://eastus.login.windows-ppe.org/17b189bc-2b81-4ec5-aa51-3e628cbc931b/oauth2/v2.0/token",
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/ClientCredentialWithRegionTests.cs:343]: That's quite subtle ... at least add a comment stating that `login.windows-ppe.org` is not known to MSAL.

**Review by bgavrilMS — APPROVED** [review]

**Review by pmaytak — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]

**Pull request merged**